docs/enhancement-proposals/off-ep-0002.md

```diff
@@ -0,0 +1,85 @@
+# OFF-EP 0002 — Support vdW interactions with a hard cut-off
+
+**Status:** Proposed
+
+**Authors:** Matt Thompson
+
+**Stakeholders:** David Mobley, John Chodera, Jeffrey Wagner, Simon Boothroyd
+
+**Acceptance criteria:** <Unanimity>
+
+**Created:** 2021-11-23
+
+**Discussion:** [Issue #7](https://github.com/openforcefield/standards/issues/7)
+
+**Implementation:** [PR #22](https://github.com/openforcefield/standards/pull/22)
+
+## Abstract
+
+This OFF-EP adds a `none` option in the `long_range_dispersion` option in the vdW section of the
+SMIRNOFF specification.
+
+## Motivation and Scope
+
+In most use cases involving force fields, vdW interactions are truncated at cutoff distance and have
+no effect past that cutoff. There [are ways](https://pubs.acs.org/doi/10.1021/jp980939v) to
+approximately correct for this, known as long-range dispersion corrections, that can improve the
+accuracy and/or speed of simulations. The SMIRNOFF specification currently has an option that
+encodes this: `long_range_dispersion`, with a default value of `"isotropic"`. There are no other
+supported values and no way to bypass adding this correction. While potentially dubious for use in
+scientific applications, users may wish to not include this correction.
+
+This OFF-EP adds `none` to the list of allowed values for the `long_range_dispersion` tag in the
+`<vdW>` section. This option communicates that there should be no correction term added to the vdW interactions
+to account for long-range dispersion forces. It does not make it the default value, so existing
+SMIRNOFF force fields and implementations should not be affected. It leaves unresolved questions
+around if a switching function or other tail correction method should be applied.
+
+## Usage and Impact
+
+This option can be used by including `long_range_dispersion="none"` in the header of the `<vdW>`
+section of a SMIRNOFF force field file. Implementations of the SMIRNOFF specification should account
+for this optional behavior. This option can be toggled by OpenMM, GROMACS, Amber, and likely other
+simulation engines.
+
+Given that long-range dispersion corrections are generally beneficial to turn on and are commonly
+used, `long_range_dispersion="none"` will likely not be turned on by most users. This proposal does
+not change the default value (`long_range_dispersion="isotropic"`). Therefore, the impact on most
+users should be none because the usage should be low. This option may make it easier for users to
+develop force fields if they desire to not include the long-range dispersion correction, which is
+currently required. Because the correction, however small, is applied over the entire range `0 < r <
```

Comment:
"Because the correction, however small, is applied over the entire range `0 < r < r_cutoff`, it may enable more accurate testing of different SMIRNOFF implementations."

This is nonsensical. Can you clarify what is meant? Are you referring to a switching function, rather than a long-range correction?

Reply:
The scope here is only the long-range correction, not the switching function. As far as I can tell from reading the papers/docs, the long-range correction is applied over the entire range of `r`, not just something past/around the cutoff distance. So two atoms at a distance `0 < r < r_cutoff` are not strictly subject to the textbook 12-6 LJ potential, but `U_{full}` from [your paper](https://pubs.acs.org/doi/10.1021/jp0735987). This makes it harder to test outside of the current OpenMM framework - i.e. by hand/one-off scripts or with other engines - since the correction term would need to be computed. The whole idea here is to expose an option to turn that off.

Reply:
I still don't understand this explanation.

A switching function is used to ensure energy and force continuity (and continuity of additional derivatives formally required by MD integrators). When computing the long-range analytical dispersion correction, the dispersion correction is usually formulated to include the remainder of the 12-6 Lennard-Jones interaction that is dampened by the switch. I think what you're trying to say is that the long-range dispersion correction _may include contributions to the integral over the range [r_switch, r_cutoff], where r_switch is the distance at which the switching function begins to dampen the direct-space LJ interactions._

Reply:
I will quote directly from the [OpenMM docs](http://docs.openmm.org/latest/api-python/generated/openmm.openmm.NonbondedForce.html#openmm.openmm.NonbondedForce.setUseDispersionCorrection) as I'm not effectively the scope here.


> When using a cutoff, by default Lennard-Jones interactions are sharply truncated at the cutoff distance. Optionally you can instead use a switching function to make the interaction smoothly go to zero over a finite distance range. To enable this, call setUseSwitchingFunction(). You must also call setSwitchingDistance() to specify the distance at which the interaction should begin to decrease. The switching distance must be less than the cutoff distance.


This proposal does not suggest any modifications that would allow a user to toggle off/tinker with the switching function, though I hope it will be addressed in in other OFF-EPs.

> Another optional feature of this class (enabled by default) is to add a contribution to the energy which approximates the effect of all Lennard-Jones interactions beyond the cutoff in a periodic system. When running a simulation at constant pressure, this can improve the quality of the result. Call setUseDispersionCorrection() to set whether this should be used.

As I understand this, it's a correction for ANY truncation method - switching, hard-cutoff, etc. - between the cutoff distance. If a switching function is not used, or something else was used in its place, it's still a term that's added to the pair potential on top of the 12-6 terms.

This correction is applied by default and turning it off is not allowed under the current SMIRNOFF spec. This proposal adds an option to turn it off, that's it. The only change in implementation would be the one-liner `setUseDispersionCorrection(False)` if the option is toggled as such.